File: TextosJapones/src/main/java/org/jisho/textosJapones/model/entities/DadosConexao.java
package org.jisho.textosJapones.model.entities;

import org.jisho.textosJapones.model.enums.Conexao;

public class DadosConexao {
    private final Long id;
    private final Conexao tipo;
    private final String url;
    private final String base;

    private final String usuario;
    private final String senha;

    public Long getId() {
        return id;
    }

    public Conexao getTipo() {
        return tipo;
    }

    public String getUrl() {
        return url;
    }

    public String getBase() {
        return base;
    }

    public String getUsuario() {
        return usuario;
    }

    public String getSenha() {
        return senha;
    }

    public DadosConexao(Long id, Conexao tipo, String url, String base, String usuario, String senha) {
        this.id = id;
        this.tipo = tipo;
        this.url = url;
        this.base = base;
        this.usuario = usuario;
        this.senha = senha;
    }
}

File: ProcessaTexto/src/main/kotlin/br/com/fenix/processatexto/model/entities/DadosConexao.kt
package br.com.fenix.processatexto.model.entities

import br.com.fenix.processatexto.model.enums.Conexao
import br.com.fenix.processatexto.model.enums.Driver
import jakarta.persistence.*

@jakarta.persistence.Entity
@Table(name = "conexoes")
data class DadosConexao(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID", nullable = false, unique = true, length = 11)
    private val id: Long?,
    @Enumerated(EnumType.STRING)
    val tipo: Conexao,
    @Column(length = 100, nullable = true)
    val url: String,
    @Column(name = "base", length = 100, nullable = true)
    val base: String,
    @Column(name = "username", length = 250, nullable = true)
    val usuario: String,
    @Column(name = "password", length = 250, nullable = true)
    val senha: String,
    @Enumerated(EnumType.STRING)
    val driver: Driver
) : EntityBase<Long?, DadosConexao>() {
    constructor(url: String, base: String, usuario: String, senha: String) :
            this(0, Conexao.PROCESSA_TEXTO, url, base, usuario, senha, Driver.MYSQL)

    constructor() : this(0, Conexao.PROCESSA_TEXTO, "", "", "", "", Driver.MYSQL)

    override fun getId(): Long? = id

    override fun create(id: Long?): DadosConexao {
        TODO("Not yet implemented")
    }
}